cluster/gce/manifests/etcd.manifest

```diff
@@ -64,13 +64,17 @@
       }
     ],
     "livenessProbe": {
-      "httpGet": {
-        "host": "127.0.0.1",
-        "port": {{ etcd_livenessprobe_port }},
-        "path": "/health"
+      "exec": {
+        "command": [
+          "/bin/sh",
+          "-c",
+          "exec /usr/local/bin/etcdctl --endpoints=127.0.0.1:{{ port }} {{ etcdctl_cerds }} --command-timeout=30s endpoint health"
```

Comment:
30s timeout seems pretty high to me

Given that processing requests generally is subseconds, the fact we have conccurrent reads (that don't block us for long time) etc. I would actually try to go lower than that.

Is it blocked by defrag as an example?
Or are you trying to accomodate only for "overload" (if the latter, there is 5k in-flight limit anyway, so you will get 429 or whatever that is in such case). There is still a case of cpu-starvation though...

Reply:
The reasoning behind 30s is that this is high enough so that if we don't receive answer by that time, it's unlikely we will ever receive any answer.

In current 5k performance tests we are seeing latency of ~5s when the etcd is overloaded so I wanted to put the thresholds significantly higher to avoid killing etcd even if it's more overloaded (if we are seeing successful responses with 5s latency, I can imagine that in some other overload scenario we will see e.g. 10s latency).

It is blocked by defrag (it was also before): https://etcd.io/docs/v3.4.0/op-guide/maintenance/#defragmentation

What value are you proposing instead?

Reply:
I was thinking about 5-10s. But given what you wrote above 5s is definitely too low.

How about 15s? If we're seeing 5s in overloaded cases, the 3x margin seems relatively safe maybe?

Reply:
15s sounds reasonable to me. I will change that